src/js/facility-locator/components/Listbox.jsx

```diff
@@ -0,0 +1,161 @@
+          );
+        })
+        }
+      </ul>
+    );
+  }
+
+  renderSelectOptionWithIcon = (facilityType, index) => {
+    const defaultSelected = !this.props.currentQuery.facilityType && !facilityType;
+    const isSelected = this.isSelectedOption((facilityType), 'facilityType');
+    const isHovered = isSelected || defaultSelected;
+    const facilityOptionClasses = classNames({
+      'dropdown-option': true,
+      'facility-option': true,
+      'is-hovered': isHovered
+    });
+    const facilityIconClasses = classNames({
+      legend: true,
+      spacer: !facilityType,
+      [`${kebabCase(facilityType)}-icon`]: facilityType
+    });
+
+    return (
+      <li
+        role="option"
+        tabIndex={this.props.dropdownActive ? '0' : '-1'}
+        aria-selected={isHovered}
+        ref={ elem => { this.props.options[index] = elem; }}
+        id={index > -1 ? (facilityType || 'AllFacilities') : null}
+        className={facilityOptionClasses}
+        onClick={() => this.props.handleFilterSelect(facilityType, 'facility')}
+        onKeyDown={this.props.navigateDropdown}>
+        <span className="flex-center">
+          <span className={facilityIconClasses}></span>
+          {facilityTypes[facilityType] || 'All Facilities'}
+        </span>
+      </li>
+    );
+  }
+
+  renderSelectButtonWithIcon = (facilityType) => {
+    const facilityOptionClasses = classNames({
+      'dropdown-option': true,
+      'facility-option': true
+    });
+    const facilityIconClasses = classNames({
+      legend: true,
+      spacer: !facilityType,
+      [`${kebabCase(facilityType)}-icon`]: facilityType
+    });
+
+    return (
+      <div
+        id="facilityDropdown"
+        className={facilityOptionClasses}
+        onKeyDown={this.props.navigateDropdown}>
+        <span className="flex-center">
+          <span className={facilityIconClasses}></span>
+          {facilityTypes[facilityType] || 'All Facilities'}
+        </span>
+      </div>
+    );
+  }
+
+  renderSelectOption = (serviceType, isMobile) => {
+    const serviceOptionClasses = classNames({
+      'dropdown-option': true,
+      'facility-option': true
+    });
+    return (
+      <div className="flex-center">
+        <div
+          id="serviceDropdown"
+          className={serviceOptionClasses}>
+          <span className="flex-center">
+            <span className="legend spacer"></span>
+            {truncate((benefitsServices[serviceType] || serviceType || 'All'), { length: (isMobile ? 38 : 27) })}
+          </span>
+        </div>
+      </div>
+    );
+  }
+
+  render() {
+    const { hasIcons, isMobile, isDisabled } = this.props;
+    const { currentQuery: { facilityType, serviceType } } = this.props;
+    return (
+      <div>
+        {hasIcons &&
+          <div>
+            <div className="flex-center">
+              {this.renderSelectButtonWithIcon(facilityType)}
+            </div>
+            <ul
+              role={!isDisabled ? 'listbox' : undefined}
```

Comment:
Does the role need to be toggled? That seems odd.

Reply:
You're right, this component should't be rendered if isDisabled is true.